task/golang-build/OWNERS

```diff
@@ -0,0 +1,4 @@
+approvers:
+- vdemeester
+reviewers:
+- vdemeester
```

Comment:
should we add all the maintainers of current catalog here?

Reply:
Yeah, I did some further investigation. For reviewers we just need to add the name to the org.yaml file in the community repo: https://github.com/tektoncd/community/pull/933.

For the approvers we need both the name in the  org.yaml file and in the `approvers` list in this `OWNERS` file. So I removed the `reviewers` list in this file and added all maintainers to the `approvers` list. PTAL 😁